### PR TITLE
Beepsky nulls

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
The Syndicate is tired of their gear falling into Nanotrasen hands. When  NULL_ENTRY crates are ordered, the Lone Operative event becomes slightly more likely.

Additionally, the syndicate may send their most deadly enforcers inside the crates instead of loot. The first crate has a 0% chance for this, but every time crates are ordered, the chance goes up by 1% per crate. The chance applies to each item, but onlly one beepsky can spawn per crate, and once one spawns, the crate stops spawning items

## Why It's Good For The Game
Improve don't remove: solves the nullcrate party issue without removing the crates or the utility they give to individual antagonists.

also, a way to get a General Beepsky without needing 2.25 uplinks

## Changelog
:cl:
balance: Nullcrates now have a downside other than an easily-gameable cost, which can also be used against the crew by a creative traitor
/:cl:

### CODE REVIEW

**Comment by dinnerwoerror** [comment]
alright zesko i already asked you this, what the hell are you smoking

why in the seven hells should null crates have a 10% chance of spawning with an emagged general beepsky in them

also, in case you didn't read the description for null crates, it's not the syndicate sending their surplus gear to a nanotrasen station for their supposed agents on board, it's NT sending their unused IAA gear to a nanotrasen station, because we all know how big brained the people behind IAAs were

**Comment by zeskorion** [comment]
> alright zesko i already asked you this, what the hell are you smoking
> 
> why in the seven hells should null crates have a 10% chance of spawning with an emagged general beepsky in them
> 
> also, in case you didn't read the description for null crates, it's not the syndicate sending their surplus gear to a nanotrasen station for their supposed agents on board, it's NT sending their unused IAA gear to a nanotrasen station, because we all know how big brained the people behind IAAs were

you're... not correct at all there. and you didn't read the mechanics behind this at all either.

the description quite literally refers to you as a syndicate agent- IAA is just a haphazard flavoring for "traitors but u kill each other"- it's even noted as a comandeered uplink in the flavor text for them

**Comment by dinnerwoerror** [comment]
you know what, never mind, i wanna see how well this goes myself

**Comment by dinnerwoerror** [comment]
ok well, to provide some constructive criticism

the idea of the chance of the lone operative event increasing is an OK idea

however, literally and figuratively slipping diarrhea into my jelly donut is not an OK idea

what kind of supplies company would slip in robotic guards that are designed to attack whoever opens the crate

bro, that company would be posting cringe, they're gonna lose reddit karma and subscribers

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -168,27 +168,38 @@
...
+			var/category = pick(uplink_items)
+			var/item = pick(uplink_items[category])
+			var/datum/uplink_item/I = uplink_items[category][item]
+			if(!I.surplus_nullcrates || prob(100 - I.surplus_nullcrates))
+				continue
+			if(crate_value < I.cost)
+				continue
+			crate_value -= I.cost
+			new I.item(C)
+	beepsky_chance += 10 //a maximum of 9 nullcrates full, if you're really lucky
```
Comment (MarkSuckerberg) [inline code/modules/cargo/packs.dm]: Does it have to increment by this much each time without going down? I don't think this is balanced as-is.

Comment (dinnerwoerror) [reply code/modules/cargo/packs.dm]: yea, i second this, it literally says in the greyed out description that it basically puts a hard cap of 9 crates per round

Comment (zeskorion) [reply code/modules/cargo/packs.dm]: I can slightly change the mechanism it works by to make it a bit more random.

Comment (MarkSuckerberg) [reply code/modules/cargo/packs.dm]: That would be nice. I'll merge it after that

Comment (zeskorion) [reply code/modules/cargo/packs.dm]: aight, changed it

**Pull request merged**